quality_control_stock_oca/__manifest__.py

```diff
@@ -5,13 +5,13 @@
 
 {
     "name": "Quality control - Stock (OCA)",
-    "version": "16.0.1.0.0",
+    "version": "17.0.1.0.0",
     "category": "Quality control",
     "license": "AGPL-3",
     "author": "OdooMRP team, AvanzOSC, Serv. Tecnol. Avanzados - Pedro M. Baeza, "
     "Agile Business Group, Odoo Community Association (OCA)",
     "website": "https://github.com/OCA/manufacture",
-    "depends": ["quality_control_oca", "stock"],
+    "depends": ["quality_control_oca", "stock", "sale_management"],
```

Comment:
Why the introduction of this dependency?

Reply:
New field sale_line_warn in product_template https://github.com/odoo/odoo/blob/c25013c79cf315856799ec009969c083d0c63965/addons/sale/models/product_template.py#L19
and needs to depends of sale module

Reply:
That's because you are testing it when the module is already installed. It's not correct. Please remove the dependency, and don't add these dependencies because you retest the module being already installed. You must always uninstall it and test on installation only, or just add this dependency as "temporary hack", but finally removing it for the final shot.

Reply:
How can add a temporary depency only for testing?

Reply:
I mean in your local. You add the dependency, but before committing, you remove it.

Reply:
Yes, but in runboat the test will fail no?

Reply:
No, as they are tested on installation. Try it... (and it's not on runboat, but on GitHub actions).

This is not your fault though, it's a flaw of the ORM and testing system, so we have to adapt to it, but we can't put any possible dependency that adds a required field in any arbitrary model.

Reply:
Just understand, sorry for the confusion hehe

Reply:
As extended information, this happens because when a module like `sale_management` is already installed, the DB not null constraint is already put in place, but on the test phase, the Python code for the default value of `sale_management` is not yet loaded. The only way to avoid this problem is to test at the same time that the module is installed, so:

- or `sale_management` is still not installed, so you don't have the DB constraint.
- or `sale_management` is installed, but both the DB constraint and the default value are in place.

Reply:
Very good information about it, many thanks for the extra help!